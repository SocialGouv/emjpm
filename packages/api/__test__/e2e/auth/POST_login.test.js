//

const request = require("supertest");
const server = require("@emjpm/api/app");
const knex = require("@emjpm/api/db/knex");

//

afterAll(async () => {
  await knex.destroy();
});

test.each`
  username       | password        | type
  ${"jeremy"}    | ${"johnson123"} | ${"individuel"}
  ${" jeRemY  "} | ${"johnson123"} | ${"individuel"}
  ${"kelly"}     | ${"bryant123"}  | ${"prepose"}
  ${"service1"}  | ${"service1"}   | ${"service"}
  ${"admin"}     | ${"admin"}      | ${"admin"}
  ${"ti1"}       | ${"ti1"}        | ${"ti"}
`("'$username' should login as $type", async ({ username, password }) => {
  const response = await request(server)
    .post("/auth/login")
    .send({
      username,
      password
    });
  expect(response.body).toMatchSnapshot({ token: expect.any(String) });
  expect(response.status).toBe(200);
});

test.each`
  username      | password         | type
  ${"inactive"} | ${"inactive"}    | ${"an inactive user"}
  ${"michael"}  | ${"jackson"}     | ${"an unregistered user"}
  ${"jeremy"}   | ${"johnson1234"} | ${"a registered user with wrong password"}
`("'$username' not login $type", async ({ username, password }) => {
  const response = await request(server)
    .post("/auth/login")
    .send({
      username,
      password
    });
  expect(response.body).toMatchSnapshot();
  expect(response.status).toBe(401);
});
