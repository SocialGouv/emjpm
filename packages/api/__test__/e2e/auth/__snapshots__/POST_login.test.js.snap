// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`' jeRemY  ' should login as individuel 1`] = `
Object {
  "success": true,
  "token": Any<String>,
  "url": "/mandataires",
}
`;

exports[`'admin' should login as admin 1`] = `
Object {
  "success": true,
  "token": Any<String>,
  "url": "/admin",
}
`;

exports[`'inactive' not login an inactive user 1`] = `
Object {
  "message": "User not found",
  "success": false,
}
`;

exports[`'jeremy' not login a registered user with wrong password 1`] = `
Object {
  "message": "User not found",
  "success": false,
}
`;

exports[`'jeremy' should login as individuel 1`] = `
Object {
  "success": true,
  "token": Any<String>,
  "url": "/mandataires",
}
`;

exports[`'kelly' should login as prepose 1`] = `
Object {
  "success": true,
  "token": Any<String>,
  "url": "/mandataires",
}
`;

exports[`'michael' not login an unregistered user 1`] = `
Object {
  "message": "User not found",
  "success": false,
}
`;

exports[`'service1' should login as service 1`] = `
Object {
  "success": true,
  "token": Any<String>,
  "url": "/services",
}
`;

exports[`'ti1' should login as ti 1`] = `
Object {
  "success": true,
  "token": Any<String>,
  "url": "/tis",
}
`;
